base/bundle.cpp

```diff
@@ -0,0 +1,119 @@
+﻿
+#include "base/bundle.hpp"
+
+#include <functional>
+#include <list>
+
+#include "base/map_util.hpp"
+#include "base/status.hpp"
+
+namespace principia {
+namespace base {
+
+thread_local std::function<bool()> AbortRequested = [] { return false; };
+
+Bundle::Bundle(int threads)
+    : terminate_on_empty_(false),
+      max_workers_(threads),
+      master_abort_(&AbortRequested) {
```

Comment:
I am not sure how this works.  This is evaluated on the thread that creates the `Bundle`.  Later, it is called on a worker thread.  A comment would be good to explain how the two get connected.